tests/phpunit/test-wc-stripe-helper.php

```diff
@@ -507,4 +507,42 @@ public function test_add_stripe_methods_in_woocommerce_gateway_order() {
 		$this->assertTrue( $gateway_order['stripe'] < $gateway_order['stripe_affirm'] );
 		$this->assertTrue( $gateway_order['stripe_affirm'] < $gateway_order['cheque'] );
 	}
+
+	/**
+	 * Test for `add_mandate_data`.
+	 *
+	 * @param string $index The index of the server variable to set.
+	 * @param string $value The value to set the server variable to.
+	 * @param string $expected The expected IP address.
+	 * @dataProvider provider_test_add_mandate_data
+	 * @return void
+	 */
+	public function test_add_mandate_data( $index, $value, $expected ) {
+		unset( $_SERVER['REMOTE_ADDR'] );
+		unset( $_SERVER['HTTP_X_REAL_IP'] );
+		unset( $_SERVER['HTTP_X_FORWARDED_FOR'] );
+
+		$_SERVER[ $index ] = $value;
+		$request           = WC_Stripe_Helper::add_mandate_data( [] );
+		$ip_address        = $request['mandate_data']['customer_acceptance']['online']['ip_address'];
```

Comment:
Nit: Given that this is a unit test, it's OK if this falls over with a fatal exception, but I would prefer to make sure that a failure is reported directly rather than indirectly.
```suggestion
		$this->assertTrue( isset( $request['mandate_data']['customer_acceptance']['online']['ip_address'] ) );
		$ip_address        = $request['mandate_data']['customer_acceptance']['online']['ip_address'];
```